src/pyFAI/gui/dialog/DetectorSelectorDialog.py

```diff
@@ -139,18 +136,71 @@ def __init__(self, parent=None):
         self.__pixelHeight.changed.connect(self.__customDetectorChanged)
 
         self._initOrientation()
+        self._initSensor()
         # By default select all the manufacturers
         self.__selectAllRegistreredDetector()
 
     def _initOrientation(self):
-        for item in pyFAI.detectors.orientation.Orientation:
+        for item in detectors.orientation.Orientation:
             if item.available:
                 self._detectorOrientation.addItem(f"{item.value}: {item.name} ", userData=item)
 
         self._detectorOrientation.currentIndexChanged.connect(self.__orientationChanged)
-        default = pyFAI.detectors.orientation.Orientation(3)
+        default = detectors.orientation.Orientation(3)
         self._detectorOrientation.setCurrentIndex(self._detectorOrientation.findData(default))
 
+    def _initSensor(self):
+        "populate the comboBox with all available sensor materials"
+        self._sensorGroup.setEnabled(False)
+        self._detectorSensorParallax.checkStateChanged.connect(self.__sensorParallax)
+        for key, value in detectors.sensors.ALL_MATERIALS.items():
+            self._detectorSensorMaterials.addItem(key, userData=value)
+        self._detectorSensorThickness.setValidator(qt.QDoubleValidator(0.0, 1e4, 1))
```

Comment:
Using a `QDoubleSpinBox` would allow to configure the widget in the designer

Reply:
Up to you, you can input the value with the keyboard as well.
It's just the default widget for `float`.